File: 10. Capitalize first and last character of each word/code.cpp
#include <iostream>
#include <cctype>
using namespace std;

void capitalize(string& str) {
    int size = str.length();
    for (int i = 0; i < size; i++) {
        if (i == 0 || i == (size - 1)) {
            if (islower(str[i]))
                str[i] = toupper(str[i]);
        } else if (str[i] == ' ') {
            if (islower(str[i - 1]))
                str[i - 1] = toupper(str[i - 1]);
            if (islower(str[i + 1]))
                str[i + 1] = toupper(str[i + 1]);
        }
    }
}

int main() {
    string str;
    cout << "Enter a string: ";
    getline(cin, str);
    capitalize(str);
    cout << "String after capitalizing the first and last letter of each word: " << endl;
    cout << str << endl;
    return 0;
}

File: 10. Capitalize first and last character of each word/code.java
import java.util.Scanner;

public class Main {
    public static void capitalize(StringBuilder str) {
        int size = str.length();
        for (int i = 0; i < size; i++) {
            if (i == 0 || i == (size - 1)) {
                if (Character.isLowerCase(str.charAt(i)))
                    str.setCharAt(i, Character.toUpperCase(str.charAt(i)));
            } else if (str.charAt(i) == ' ') {
                if (Character.isLowerCase(str.charAt(i - 1)))
                    str.setCharAt(i - 1, Character.toUpperCase(str.charAt(i - 1)));
                if (Character.isLowerCase(str.charAt(i + 1)))
                    str.setCharAt(i + 1, Character.toUpperCase(str.charAt(i + 1)));
            }
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a string: ");
        StringBuilder str = new StringBuilder(scanner.nextLine());
        capitalize(str);
        System.out.println("String after capitalizing the first and last letter of each word: ");
        System.out.println(str);
    }
}
